src/charm.py

```diff
@@ -224,27 +242,37 @@ def get_env_vars(self) -> dict:
 
         return env_vars
 
-    def _update_workload_container_config(self, event):
-        """Update workload with all available configuration data."""
+    def _update_workload_container_config(self, event: Optional[HookEvent]):
+        """
+        Update workload with all available configuration data.
+
+        Note that given event should be deferrable. For example, action events
+        (of type ActionEvent), will raise exception if their `defer` method is
+        invoked. So, the caller of this method should pass event as None if it's
+        not a deferrable event.
+        """
         if not self._state.is_ready():
-            event.defer()
+            if event:
+                event.defer()
```

Comment:
What about wrapping this in a function like `try_defer(event)`? or even just `defer(event)`

Reply:
Yeah, why not.